main.tf

```diff
@@ -0,0 +1,79 @@
+resource "aws_cloudtrail" "trail" {
+  name           = "trail${replace(title(var.project), " ", "")}${title(var.environment)}"
+  s3_bucket_name = "${var.s3_bucket_name}"
+
+  enable_logging                = true
+  include_global_service_events = true
+  is_multi_region_trail         = false
+  enable_log_file_validation    = false
+
+  tags {
+    Name        = "${var.project}"
+    Environment = "${var.environment}"
+  }
+
+  depends_on = ["aws_s3_bucket.trail"]
+}
+
+resource "aws_s3_bucket" "trail" {
+  count = "${var.create_s3_bucket ? 1 : 0}"
+
+  bucket = "${var.s3_bucket_name}"
+  region = "${var.region}"
+
+  force_destroy = true
+
+  lifecycle_rule {
```

Comment:
Interesting. I'm not sure that my default approach would be to prune these logs. Possibly the more conservative approach would be to change their storage class? Not sure if there is a way to use one vs. the other at the same time.

Reply:
I like that idea! To be clear, I'm interpreting you to be suggesting that we default to changing the storage class of old logs, but allow the user to optionally enable expiration, too.

Normally we wouldn't be able to conditionally enable these blocks due to [Terraform not allowing conditional inline blocks](https://github.com/azavea/terraform-aws-cloudtrail/pull/1/files#r238788330), but in this case we can hack around it thanks to the fact that [`lifecycle_rule` blocks have an `enabled` flag](https://stackoverflow.com/a/42468226) that can be conditionally flipped on or off based on the value of an input variable.